CHANGELOG.md

```diff
@@ -6,6 +6,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+### Changed
+
+- `micropip.install` now installs wheels from PyPI or custom indexes before built-in
+  Pyodide packages, reversing the previous behavior.
+  [#181](https://github.com/pyodide/micropip/pull/181)
```

Comment:
Is there a user-facing change related to this? I think the order of installation is an implementation detail that is not visible to the user. What a user sees whether the whole set of packages (with dependencies) are installed or not.

Reply:
Yeah, I don't know: seems like In interactive package management, which can be modified by about five different flags now, I think "reversing what we used to do," warrants at least a note. I am _not sure_ if it's breaking, but things folk did before might not work now, and they might not need workarounds (like the double-install thing).

Reply:
I see. Makes sense. Let me check the code more carefully and review.

I think the current wording could be a bit misleading to users though. When the same package exists in both the pyodide lock file and PyPI, the lockfile currently takes precedence (in terms of which package to choose, not the order of installation). But maybe the user will think it in a different way.

Reply:
Right: today, in the transaction phase once a dependency chain hits a pyodide package, further dependency resolution was skipped because the lockfile would take care of the rest. This was a performance win, for sure. In the install stage, they were all mixed together in `wheel_tasks`, and there was really no way to make any claims about what would happen.

With _this_ change, the transaction will keep going all the way down for every novel dependency.  In the install phase, the PyPI wheels get installed (in parallel) before calling `loadPackage`, which _won't_ re-install (and re-download) a pyodide wheel.

This _does_ mean there will be an extra `add_dependency_inner` for each `depends` item, but I couldn't figure out a way to do it that would be remotely correct.

Reply:
... but that's a lot to say in a CHANGELOG, maybe there's a terser way?

Reply:
tried some more things on [92baa16](https://github.com/pyodide/micropip/pull/181/commits/92baa16724b95b947ab9da8866a287c192578218)

Reply:
> With this change, the transaction will keep going all the way down for every novel dependency. In the install phase, the PyPI wheels get installed (in parallel) before calling loadPackage, which won't re-install (and re-download) a pyodide wheel.

Hmm, I still don't quite get what `which won't re-install` means. By default, during the dependency resolution step, micropip will check if the package is available in the lockfile, and select the one from the lockfile instead of the one from the PyPI.

So, in the installation step, we expect no overlap between `transaction.pyodide_packages` and  `transaction.wheels`. Does this PR change this behavior?

Reply:
Got it. I didn't consider putting multiple packages in `install(...)`. Thanks.